tests/integration/migration/cassandra/CassandraMigrationManagerTests.cpp

```diff
@@ -0,0 +1,314 @@
+#include "migration/impl/MigrationManagerBase.hpp"
+#include "migration/impl/MigrationManagerInterface.hpp"
+#include "migration/impl/MigratorsRegister.hpp"
+#include "util/CassandraDBHelper.hpp"
+#include "util/LoggerFixtures.hpp"
+#include "util/MockPrometheus.hpp"
+#include "util/config/Config.hpp"
+
+#include <TestGlobals.hpp>
+#include <boost/json/parse.hpp>
+#include <fmt/core.h>
+#include <gtest/gtest.h>
+#include <xrpl/basics/base_uint.h>
+
+#include <algorithm>
+#include <cstdint>
+#include <memory>
+#include <optional>
+#include <string>
+#include <utility>
+
+using namespace util;
+using namespace std;
+using namespace prometheus;
+using namespace data::cassandra;
+using namespace migration;
+namespace json = boost::json;
+
+// Register the migrators
+using CassandraSupportedTestMigrators = migration::impl::MigratorsRegister<
+    CassandraMigrationTestBackend,
+    ExampleObjectsMigrator,
+    ExampleTransactionsMigrator,
+    ExampleLedgerMigrator,
+    ExampleDropTableMigrator>;
+
+// Define the test migration manager
+using CassandraMigrationTestManager = migration::impl::MigrationManagerBase<CassandraSupportedTestMigrators>;
+
+namespace {
+std::pair<std::shared_ptr<migration::impl::MigrationManagerInterface>, std::shared_ptr<CassandraMigrationTestBackend>>
+make_MigrationTestManagerAndBackend(util::Config const& config)
+{
+    auto const cfg = config.section("database.cassandra");
+
+    auto const backendPtr = std::make_shared<CassandraMigrationTestBackend>(data::cassandra::SettingsProvider{cfg});
+
+    return std::make_pair(
+        std::make_shared<CassandraMigrationTestManager>(backendPtr, cfg.sectionOr("migration", {})), backendPtr
+    );
+}
+}  // namespace
+
+class MigrationCassandraSimpleTest : public WithPrometheus, public NoLoggerFixture {
+    // This function is used to prepare the database before running the tests
+    // It is called in the SetUp function. Different tests can override this function to prepare the database
+    // differently
+    virtual void
+    setupDatabase()
+    {
+    }
+
+protected:
+    Config cfg{json::parse(fmt::format(
+        R"JSON({{
+            "database": {{
+                "type": "cassandra",
+                "cassandra": {{
+                    "contact_points": "{}",
+                    "keyspace": "{}",
+                    "replication_factor": 1
+                }}
+            }}
+        }})JSON",
+        TestGlobals::instance().backendHost,
+        TestGlobals::instance().backendKeyspace
+    ))};
+
+    std::shared_ptr<migration::impl::MigrationManagerInterface> testMigrationManager;
+    std::shared_ptr<CassandraMigrationTestBackend> testMigrationBackend;
+
+    MigrationCassandraSimpleTest()
+    {
+        auto const testBundle = make_MigrationTestManagerAndBackend(cfg);
+        testMigrationManager = testBundle.first;
+        testMigrationBackend = testBundle.second;
+    }
+
+    void
+    SetUp() override
+    {
+        setupDatabase();
```

Comment:
The base classes don't have Setup and TearDown. But in case they add one in future, I will add the callee here.
I can not move setupDatabase to constructor, it is virtual function. I want to it behave polymorphically for different migrator test.